EqSat/src/UnionFind.cpp

```diff
@@ -24,12 +26,37 @@ Id UnionFind::find(Id id) const
     return id;
 }
 
+Id UnionFind::find(Id id)
+{
+    LUAU_ASSERT(size_t(id) < parents.size());
+
+    // An e-class id 𝑎 is canonical iff find(𝑎) = 𝑎.
+    if (id != parents[size_t(id)])
+        // Note: we don't update the ranks here since a rank
+        // represents the upper bound on the maximum depth of a tree
+        parents[size_t(id)] = find(parents[size_t(id)]);
```

Comment:
Can I see a non-recursive implementation? If the `UnionFind` have a sufficiently large amount of sets, and the stack has some indeterminate size but is close to stack overflowing, I would prefer for `UnionFind` not to be the reason to stack overflow.